rust/sdk/src/client.rs

```diff
@@ -282,86 +263,33 @@ impl EventService for ActyxClient {
         })?)
     }
 
-    /// Publishes a set of new events.
-    ///
-    /// If an authorization error (code 401) is returned, it will try to re-authenticate.
-    /// If the service is unavailable (code 503), this method will retry to perform the
-    /// request up to 10 times with exponentially increasing delay - currently,
-    /// this behavior is only available if the `with-tokio` feature is enabled.
-    async fn publish(&self, request: PublishRequest) -> anyhow::Result<PublishResponse> {
-        let body = serde_json::to_value(&request).context(|| format!("serializing {:?}", &request))?;
-        let response = self
-            .do_request(|c| c.post(self.events_url("publish")).json(&body))
-            .await?;
-        let bytes = response
-            .bytes()
-            .await
-            .context(|| format!("getting body for GET {}", self.events_url("publish")))?;
-        Ok(serde_json::from_slice(bytes.as_ref()).context(|| {
-            format!(
-                "deserializing publish response from {:?} received from GET {}",
-                bytes,
-                self.events_url("publish")
-            )
-        })?)
+    /// Returns a builder for publishing events.
+    pub fn publish(&self) -> Publish<'_> {
+        Publish::new(&self)
     }
 
-    /// Query events known at the time the request was received by the service.
-    ///
-    /// If an authorization error (code 401) is returned, it will try to re-authenticate.
-    /// If the service is unavailable (code 503), this method will retry to perform the
-    /// request up to 10 times with exponentially increasing delay - currently,
-    /// this behavior is only available if the `with-tokio` feature is enabled.
-    async fn query(&self, request: QueryRequest) -> anyhow::Result<BoxStream<'static, QueryResponse>> {
-        let body = serde_json::to_value(&request).context(|| format!("serializing {:?}", &request))?;
-        let response = self
-            .do_request(|c| c.post(self.events_url("query")).json(&body))
-            .await?;
-        let res = to_lines(response.bytes_stream())
-            .map(|bs| serde_json::from_slice(bs.as_ref()))
-            // FIXME this swallows deserialization errors, silently dropping event envelopes
-            .filter_map(|res| future::ready(res.ok()));
-        Ok(res.boxed())
+    /// Returns a builder to query events.
```

Comment:
similar here, showing two queries (one directly awaiting, the other with setting some option)